### ISSUE TITLE
[WebToolsE2E][Aspire] When running a net9.0 aspire project on a machine with only SDK 9.0 installed, it fails.

### ISSUE THREAD

**Issue by v-reinawang** [dotnet/aspire#5599]
**INSTALL STEPS**

1. Clean machine: Win11 x64 23h2 ENU
2. Install .NET SDK 9.0 RC1 [9.0.100-rc.1.24452.12]
3. Install .NET Aspire workload 9.0 P4 [9.0.0-preview.4.24456.4]

**REPRO STEPS**
1. Open cmd, run `set DOTNET_MULTILEVEL_LOOKUP=0`
2. Use the command to create an aspire project:
```
dotnet new aspire-starter -o aspire1
cd aspire1/aspire1.AppHost
```
3. Manually open the project and replace all net8.0 TFMs in the projects with net9.0 TFMs for the TargetFramework properties.
4. Build and run

**ACTUAL**
Build is fine but run fails.
![image](https://github.com/user-attachments/assets/256313d2-55b2-4fc7-9add-98fd10e20c8a)
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6040000Z You must install or update .NET to run this application.
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6040000Z
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6040000Z App: C:\Users\v-reinawang\.nuget\packages\aspire.dashboard.sdk.win-x64\9.0.0-preview.4.24456.4\tools\Aspire.Dashboard.exe
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6040000Z Architecture: x64
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6050000Z Framework: 'Microsoft.NETCore.App', version '8.0.0' (x64)
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6050000Z .NET location: C:\Program Files\dotnet
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6050000Z
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6050000Z The following frameworks were found:
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6050000Z   9.0.0-rc.1.24431.7 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6050000Z
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6050000Z Learn more:
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6050000Z https://aka.ms/dotnet/app-launch-failed
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6050000Z
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6050000Z To install missing framework, download:
fail: Aspire.Hosting.Dashboard[0]
      2024-09-09T10:24:44.6050000Z https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=8.0.0&arch=x64&rid=win-x64&os=win10

**EXPECTED**
Can run successfully.

**Comment by v-reinawang** [comment]
According to https://github.com/dotnet/aspire/pull/5540, this scenario (install only SDK 9.0 + Aspire 9.0 P4, create aspire-starter with TargetFramework of net8.0, update all TFMs from net8.0 to net9.0) can run successfully.

**Comment by joperezr** [comment]
You are correct, this is supposed to work. The reason why it didn't is because the snap for 9.0 preview 4 branch was done _**before**_ the above PR was merged. I'll open a backport and merge it so that next validation should pass.

**Comment by marinasundstrom** [comment]
When will this be fixed? I'm moving my stuff over to .NET 9.

**Comment by v-elenafeng** [comment]
@joperezr, with the Aspire 9.0 P4 release cancelled, this is no longer an issue. Should we just close on this? Will check the scenario once we have a release build for 9.0.

**Comment by joperezr** [comment]
> When will this be fixed? I'm moving my stuff over to .NET 9. This is the blocker after all work.

Aspire 9.0 will support targeting .NET 9 and .NET 8. It will get released in November along with .NET 9. We will also have a prerelease of this that should have this fix around mid October.

**Comment by joperezr** [comment]
This is fixed in RC1 and GA. Closing.

**Comment by v-reinawang** [comment]
Verify that this issue has been fixed in Aspire 9.0 GA [9.0.0-rc.2.24519.2].
